### PR TITLE
HBASE-22572 Javadoc Warnings: @link reference not found

### PR DESCRIPTION
**Pull request by syedmurtazahassan**
Fixed Javadoc Warnings: @link reference not found

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 192 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 8 new or modified test files. |
||| _ master Compile Tests _ |
| 0 | mvndep | 35 | Maven dependency ordering for branch |
| +1 | mvninstall | 259 | master passed |
| +1 | compile | 135 | master passed |
| +1 | checkstyle | 131 | master passed |
| +1 | shadedjars | 264 | branch has no errors when building our shaded downstream artifacts. |
| 0 | findbugs | 0 | Skipped patched modules with no Java source: hbase-annotations |
| +1 | findbugs | 321 | master passed |
| +1 | javadoc | 92 | master passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 15 | Maven dependency ordering for patch |
| +1 | mvninstall | 252 | the patch passed |
| +1 | compile | 129 | the patch passed |
| +1 | javac | 129 | the patch passed |
| -1 | checkstyle | 63 | hbase-server: The patch generated 2 new + 85 unchanged - 8 fixed = 87 total (was 93) |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 261 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 748 | Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2. |
| 0 | findbugs | 0 | Skipped patched modules with no Java source: hbase-annotations |
| +1 | findbugs | 347 | the patch passed |
| +1 | javadoc | 12 | hbase-annotations in the patch passed. |
| +1 | javadoc | 21 | hbase-common generated 0 new + 1 unchanged - 1 fixed = 1 total (was 2) |
| +1 | javadoc | 33 | hbase-server in the patch passed. |
| +1 | javadoc | 17 | hbase-mapreduce in the patch passed. |
| +1 | javadoc | 14 | hbase-endpoint in the patch passed. |
||| _ Other Tests _ |
| +1 | unit | 14 | hbase-annotations in the patch passed. |
| +1 | unit | 170 | hbase-common in the patch passed. |
| +1 | unit | 13745 | hbase-server in the patch passed. |
| +1 | unit | 1383 | hbase-mapreduce in the patch passed. |
| +1 | unit | 267 | hbase-endpoint in the patch passed. |
| +1 | asflicense | 175 | The patch does not generate ASF License warnings. |
| | | 19520 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-306/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/306 |
| Optional Tests |  dupname  asflicense  javac  javadoc  unit  findbugs  shadedjars  hadoopcheck  hbaseanti  checkstyle  compile  |
| uname | Linux 1135bbe78868 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | master / ab4453158a |
| maven | version: Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T18:33:14Z) |
| Default Java | 1.8.0_181 |
| findbugs | v3.1.11 |
| checkstyle | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-306/1/artifact/out/diff-checkstyle-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-306/1/testReport/ |
| Max. process+thread count | 5608 (vs. ulimit of 10000) |
| modules | C: hbase-annotations hbase-common hbase-server hbase-mapreduce hbase-endpoint U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-306/1/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

File: hbase-server/src/main/java/org/apache/hadoop/hbase/master/balancer/StochasticLoadBalancer.java [hunk 1]
```diff
@@ -28,6 +28,8 @@
 import java.util.List;
 import java.util.Map;
 import java.util.Random;
+
+import org.apache.commons.math3.stat.descriptive.DescriptiveStatistics;
```
Comment (HorizonNet) [inline hbase-server/src/main/java/org/apache/hadoop/hbase/master/balancer/StochasticLoadBalancer.java]: Please don't import a class just to resolve a link in a Javadoc.

File: hbase-server/src/test/java/org/apache/hadoop/hbase/HBaseTestCase.java [hunk 2]
```diff
@@ -173,7 +173,7 @@ protected HRegion openClosedRegion(final HRegion closedRegion)
   }
 
   /**
-   * Create a table of name <code>name</code> with {@link COLUMNS} for
+   * Create a table of name <code>name</code> with {@link #COLUMNS} for
```
Comment (HorizonNet) [inline hbase-server/src/test/java/org/apache/hadoop/hbase/HBaseTestCase.java]: Please also change `<code>` to its Javadoc version.

File: hbase-server/src/test/java/org/apache/hadoop/hbase/HBaseTestCase.java [hunk 3]
```diff
@@ -183,7 +183,7 @@ protected HTableDescriptor createTableDescriptor(final String name) {
...
```
Comment (HorizonNet) [inline hbase-server/src/test/java/org/apache/hadoop/hbase/HBaseTestCase.java]: Ditto.

File: hbase-server/src/test/java/org/apache/hadoop/hbase/HBaseTestCase.java [hunk 4]
```diff
@@ -196,7 +196,7 @@ protected HTableDescriptor createTableDescriptor(final String name,
...
```
Comment (HorizonNet) [inline hbase-server/src/test/java/org/apache/hadoop/hbase/HBaseTestCase.java]: Ditto.

**Comment by syedmurtazahassan** [comment]
Thanks for your comments. Made some changes accordingly and in between had some checkstyle issues and fixed them by adding an entry in 'checkstyle-suppressions.xml' file.

Kindly see the 3rd commit.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 46 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 1 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 8 new or modified test files. |
||| _ master Compile Tests _ |
| 0 | mvndep | 13 | Maven dependency ordering for branch |
| +1 | mvninstall | 243 | master passed |
| +1 | compile | 149 | master passed |
| +1 | checkstyle | 132 | master passed |
| +1 | shadedjars | 267 | branch has no errors when building our shaded downstream artifacts. |
| 0 | findbugs | 0 | Skipped patched modules with no Java source: hbase-checkstyle hbase-annotations |
| +1 | findbugs | 309 | master passed |
| +1 | javadoc | 110 | master passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 14 | Maven dependency ordering for patch |
| +1 | mvninstall | 241 | the patch passed |
| +1 | compile | 148 | the patch passed |
| +1 | javac | 148 | the patch passed |
| +1 | checkstyle | 135 | root: The patch generated 0 new + 100 unchanged - 7 fixed = 100 total (was 107) |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | xml | 2 | The patch has no ill-formed XML file. |
| +1 | shadedjars | 273 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 767 | Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2. |
| 0 | findbugs | 0 | Skipped patched modules with no Java source: hbase-checkstyle hbase-annotations |
| +1 | findbugs | 309 | the patch passed |
| +1 | javadoc | 12 | hbase-checkstyle in the patch passed. |
| +1 | javadoc | 13 | hbase-annotations in the patch passed. |
| +1 | javadoc | 20 | hbase-common generated 0 new + 1 unchanged - 1 fixed = 1 total (was 2) |
| +1 | javadoc | 33 | hbase-server in the patch passed. |
| +1 | javadoc | 18 | hbase-mapreduce in the patch passed. |
| +1 | javadoc | 14 | hbase-endpoint in the patch passed. |
||| _ Other Tests _ |
| +1 | unit | 12 | hbase-checkstyle in the patch passed. |
| +1 | unit | 13 | hbase-annotations in the patch passed. |
| +1 | unit | 169 | hbase-common in the patch passed. |
| +1 | unit | 14992 | hbase-server in the patch passed. |
| +1 | unit | 1428 | hbase-mapreduce in the patch passed. |
| +1 | unit | 254 | hbase-endpoint in the patch passed. |
| +1 | asflicense | 193 | The patch does not generate ASF License warnings. |
| | | 20732 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-306/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/306 |
| Optional Tests |  dupname  asflicense  javac  javadoc  unit  findbugs  shadedjars  hadoopcheck  hbaseanti  checkstyle  compile  xml  |
| uname | Linux ca8cc1bbd88c 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | master / 214553d17a |
| maven | version: Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T18:33:14Z) |
| Default Java | 1.8.0_181 |
| findbugs | v3.1.11 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-306/3/testReport/ |
| Max. process+thread count | 5478 (vs. ulimit of 10000) |
| modules | C: hbase-checkstyle hbase-annotations hbase-common hbase-server hbase-mapreduce hbase-endpoint U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-306/3/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by HorizonNet** [comment]
@syedmurtazahassan Could you please resolve the merge conflicts.

**Comment by syedmurtazahassan** [comment]
@HorizonNet I have resolved the merge conflicts. Kindly have a look when you have time.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 0 | Docker mode activated. |
| -1 | patch | 8 | https://github.com/apache/hbase/pull/306 does not apply to master. Rebase required? Wrong Branch? See https://yetus.apache.org/documentation/in-progress/precommit-patchnames for help. |


| Subsystem | Report/Notes |
|----------:|:-------------|
| GITHUB PR | https://github.com/apache/hbase/pull/306 |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-306/4/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by syedmurtazahassan** [comment]
@HorizonNet 
Solved the " #306 does not apply to master" problem. 
Kindly review.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 95 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 10 new or modified test files. |
||| _ master Compile Tests _ |
| 0 | mvndep | 37 | Maven dependency ordering for branch |
| +1 | mvninstall | 324 | master passed |
| +1 | compile | 190 | master passed |
| +1 | checkstyle | 131 | master passed |
| +1 | shadedjars | 272 | branch has no errors when building our shaded downstream artifacts. |
| 0 | findbugs | 0 | Skipped patched modules with no Java source: hbase-checkstyle hbase-annotations |
| +1 | findbugs | 392 | master passed |
| +1 | javadoc | 138 | master passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 15 | Maven dependency ordering for patch |
| +1 | mvninstall | 256 | the patch passed |
| +1 | compile | 196 | the patch passed |
| +1 | javac | 196 | the patch passed |
| +1 | checkstyle | 128 | root: The patch generated 0 new + 114 unchanged - 1 fixed = 114 total (was 115) |
| +1 | whitespace | 1 | The patch has no whitespace issues. |
| +1 | xml | 1 | The patch has no ill-formed XML file. |
| +1 | shadedjars | 271 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 823 | Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2. |
| 0 | findbugs | 0 | Skipped patched modules with no Java source: hbase-checkstyle hbase-annotations |
| +1 | findbugs | 458 | the patch passed |
| +1 | javadoc | 9 | hbase-checkstyle in the patch passed. |
| +1 | javadoc | 10 | hbase-annotations in the patch passed. |
| +1 | javadoc | 18 | hbase-common generated 0 new + 1 unchanged - 1 fixed = 1 total (was 2) |
| +1 | javadoc | 19 | hbase-client in the patch passed. |
| +1 | javadoc | 13 | hbase-zookeeper in the patch passed. |
| +1 | javadoc | 31 | hbase-server in the patch passed. |
| +1 | javadoc | 15 | hbase-mapreduce in the patch passed. |
| +1 | javadoc | 15 | hbase-endpoint in the patch passed. |
||| _ Other Tests _ |
| +1 | unit | 11 | hbase-checkstyle in the patch passed. |
| +1 | unit | 11 | hbase-annotations in the patch passed. |
| +1 | unit | 170 | hbase-common in the patch passed. |
| +1 | unit | 111 | hbase-client in the patch passed. |
| +1 | unit | 47 | hbase-zookeeper in the patch passed. |
| -1 | unit | 16255 | hbase-server in the patch failed. |
| +1 | unit | 1317 | hbase-mapreduce in the patch passed. |
| +1 | unit | 262 | hbase-endpoint in the patch passed. |
| +1 | asflicense | 250 | The patch does not generate ASF License warnings. |
| | | 22722 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.master.procedure.TestSCPWithReplicas |
|   | hadoop.hbase.master.procedure.TestProcedurePriority |
|   | hadoop.hbase.util.TestFromClientSide3WoUnsafe |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-306/6/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/306 |
| Optional Tests |  dupname  asflicense  javac  javadoc  unit  findbugs  shadedjars  hadoopcheck  hbaseanti  checkstyle  compile  xml  |
| uname | Linux bd58f3bde1ba 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | master / 0c8dc5d97e |
| maven | version: Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T18:33:14Z) |
| Default Java | 1.8.0_181 |
| findbugs | v3.1.11 |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-306/6/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-306/6/testReport/ |
| Max. process+thread count | 5278 (vs. ulimit of 10000) |
| modules | C: hbase-checkstyle hbase-annotations hbase-common hbase-client hbase-zookeeper hbase-server hbase-mapreduce hbase-endpoint U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-306/6/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by HorizonNet** [comment]
There seems to be a build running. Let's wait for the results.

**Comment by saintstack** [comment]
Patch looks nice. Re-running build.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 173 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 10 new or modified test files. |
||| _ master Compile Tests _ |
| 0 | mvndep | 24 | Maven dependency ordering for branch |
| +1 | mvninstall | 237 | master passed |
| +1 | compile | 188 | master passed |
| +1 | checkstyle | 132 | master passed |
| +1 | shadedjars | 262 | branch has no errors when building our shaded downstream artifacts. |
| 0 | findbugs | 0 | Skipped patched modules with no Java source: hbase-checkstyle hbase-annotations |
| +1 | findbugs | 401 | master passed |
| +1 | javadoc | 140 | master passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 16 | Maven dependency ordering for patch |
| +1 | mvninstall | 230 | the patch passed |
| +1 | compile | 187 | the patch passed |
| +1 | javac | 187 | the patch passed |
| +1 | checkstyle | 130 | root: The patch generated 0 new + 114 unchanged - 1 fixed = 114 total (was 115) |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | xml | 2 | The patch has no ill-formed XML file. |
| +1 | shadedjars | 265 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 717 | Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2. |
| 0 | findbugs | 0 | Skipped patched modules with no Java source: hbase-checkstyle hbase-annotations |
| +1 | findbugs | 445 | the patch passed |
| +1 | javadoc | 12 | hbase-checkstyle in the patch passed. |
| +1 | javadoc | 12 | hbase-annotations in the patch passed. |
| +1 | javadoc | 21 | hbase-common generated 0 new + 1 unchanged - 1 fixed = 1 total (was 2) |
| +1 | javadoc | 20 | hbase-client in the patch passed. |
| +1 | javadoc | 15 | hbase-zookeeper in the patch passed. |
| +1 | javadoc | 32 | hbase-server in the patch passed. |
| +1 | javadoc | 17 | hbase-mapreduce in the patch passed. |
| +1 | javadoc | 14 | hbase-endpoint in the patch passed. |
||| _ Other Tests _ |
| +1 | unit | 11 | hbase-checkstyle in the patch passed. |
| +1 | unit | 14 | hbase-annotations in the patch passed. |
| +1 | unit | 179 | hbase-common in the patch passed. |
| +1 | unit | 111 | hbase-client in the patch passed. |
| +1 | unit | 48 | hbase-zookeeper in the patch passed. |
| -1 | unit | 14328 | hbase-server in the patch failed. |
| +1 | unit | 1442 | hbase-mapreduce in the patch passed. |
| +1 | unit | 277 | hbase-endpoint in the patch passed. |
| +1 | asflicense | 259 | The patch does not generate ASF License warnings. |
| | | 20768 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.client.TestAdmin2 |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-306/7/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/306 |
| Optional Tests |  dupname  asflicense  javac  javadoc  unit  findbugs  shadedjars  hadoopcheck  hbaseanti  checkstyle  compile  xml  |
| uname | Linux ac36879afc99 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | master / 52eb893bcc |
| maven | version: Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T18:33:14Z) |
| Default Java | 1.8.0_181 |
| findbugs | v3.1.11 |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-306/7/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-306/7/testReport/ |
| Max. process+thread count | 5478 (vs. ulimit of 10000) |
| modules | C: hbase-checkstyle hbase-annotations hbase-common hbase-client hbase-zookeeper hbase-server hbase-mapreduce hbase-endpoint U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-306/7/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by saintstack** [comment]
Merged. Failed test unrelated to a bunch of javadoc changes.

**Pull request merged**